bluesky/callbacks/mpl_plotting.py

```diff
@@ -233,12 +236,28 @@ def event(self, doc):
         super().event(doc)
 
     def update(self, x, y, I):
+        # if one is None all are
+        if self._minx is None:
+            self._minx = x
+            self._maxx = x
+            self._miny = y
+            self._maxy = y
+
         self._xdata.append(x)
         self._ydata.append(y)
         self._Idata.append(I)
         offsets = np.vstack([self._xdata, self._ydata]).T
         self.sc.set_offsets(offsets)
         self.sc.set_array(np.asarray(self._Idata))
+
+        if self.xlim is None:
+            minx, maxx = np.minimum(x, self._minx), np.maximum(x, self._maxx)
+            self.ax.set_xlim(minx, maxx)
```

Comment:
@tacaswell I tried different functions that appeared to automatically reset limits, i could not get them to work correctly, so i have left a manual limit setting per event for now.

Reply:
I believe the auto-limiters include nice margins, which we are forgoing here. Can you open an issue to revisit this and hopefully use matplotlib's relimiter in the future? Doesn't have to block this PR.

Reply:
Did you use https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.relim.html?highlight=relim#matplotlib.axes.Axes.relim ?

Reply:
@danielballan yes i thought of writing my own, but then realized, sometimes we may want an offset, sometimes a percentage. ill try `relim` again just in case, but yes i did. ill get back to you. thanks!

Reply:
It still doesn't work. I'm in interacitve mode. Here's code that can repdroduce this:
```py
from pylab import *
ion()
clf();plot([1,2,3,], [4,5,6])
xlim(0,1)
gca().relim()
```

Reply:
(does not change limits back to normal). Is this potentially a bug? (or my poor usage?)

Reply:
I can reproduce that. I checked `gca()._stale == False`; it does not seem to be a drawing issue.

Reply:
That is the expected behavior `relim` computes the new data limits, but does not change them.  You need to call `autoscale_view`. 

However, reading the docstring it does not take into account `Collection` instances.  Do not rember the details of why off the top of my head though.